src/Compilers/VisualBasic/Test/Emit/Emit/DynamicAnalysis/DynamicInstrumentationTests.vb

```diff
@@ -0,0 +1,800 @@
+MyLabel:
+        Do
+            Exit Do
+            y += 1
+        Loop
+        For x As Integer = 1 To 10
+            Exit For
+            y += 1
+        Next
+        Try
+            Exit Try
+            y += 1
+        Catch ex As System.Exception
+        End Try
+        Select Case y
+            Case 0
+                Exit Select
+                y += 0
+        End Select
+        If y = 0 Then
+            Exit Sub
+        End If
+        GoTo MyLabel
+    End Sub
+
+    Public Sub Main(args As String())                                       ' Method 2
+        Branches()
+        Microsoft.CodeAnalysis.Runtime.Instrumentation.FlushPayload()
+    End Sub
+End Module
+]]>
+                                         </file>
+            Dim source As Xml.Linq.XElement = <compilation></compilation>
+            source.Add(testSource)
+            source.Add(InstrumentationHelperSource)
+
+            Dim expectedOutput As XCData = <![CDATA[
+Flushing
+1
+True
+True
+True
+False
+True
+True
+False
+True
+False
+True
+True
+False
+True
+True
+False
+2
+True
+True
+True
+5
+True
+True
+False
+True
+True
+True
+True
+True
+True
+True
+True
+]]>
+
+            CompileAndVerify(source, expectedOutput)
+        End Sub
+
+        <Fact>
+        Public Sub MissingMethodNeededForAnaysis()
+            Dim testSource As XElement = <file name="c.vb">
+                                             <![CDATA[
+Namespace System
+    Public Class [Object] : End Class
+    Public Structure Int32 : End Structure
+    Public Structure [Boolean] : End Structure
+    Public Class [String] : End Class
+    Public Class Exception : End Class
+    Public Class ValueType : End Class
+    Public Class [Enum] : End Class
+    Public Structure Void : End Structure
+    Public Class Guid : End Class
+End Namespace
+
+Namespace System
+    Public Class Console
+        Public Shared Sub WriteLine(s As String)
+        End Sub
+        Public Shared Sub WriteLine(i As Integer)
+        End Sub
+        Public Shared Sub WriteLine(b As Boolean)
+        End Sub
+    End Class
+End Namespace
+
+Class Program
+    Public Shared Sub Main(args As String())
+        TestMain()
+        Microsoft.CodeAnalysis.Runtime.Instrumentation.FlushPayload()
+    End Sub
+    
+    Shared Sub TestMain()
+    End Sub
+End Class
+]]>
+                                         </file>
+            Dim source As Xml.Linq.XElement = <compilation></compilation>
+            source.Add(testSource)
+            source.Add(InstrumentationHelperSource)
+
+            Dim diagnostics As ImmutableArray(Of Diagnostic) = CreateCompilation(source).GetEmitDiagnostics(EmitOptions.Default.WithInstrument("Test.Flag"))
+            For Each Diagnostic As Diagnostic In diagnostics
+                If Diagnostic.Code = ERRID.ERR_MissingRuntimeHelper AndAlso Diagnostic.Arguments(0).Equals("System.Guid.Parse") Then
+                    Return
+                End If
+            Next
```

Comment:
Yes, but that would be more confusing than helpful. There are a bunch of irrelevant-for-the-purposes-of-this-test diagnostics (due to various other missing things in the fake System namespace). The important point is that the absence of at least one helper needed for synthesizing the instrumentation stuff is diagnosed without crashing the compiler.